docker/nginx/lang.lua

```diff
@@ -0,0 +1,83 @@
+--------------------------------------------------
+-- HTTP Accept-Language header handler          --
+-- @author f.ghibellini@gmail.com               --
+--         since altered by Mozilla             --
+-- @license MIT                                 --
+-- @requires:                                   --
+--  -gnu find                                   --
+-- @description:                                --
+--     redirects to subfolders according        --
+--     to the http Accept-Language header       --
+-- @example configuration:                      --
+--                                              --
+--     server {                                 --
+--         listen 8080 default_server;          --
+--         index index.html index.htm;          --
+--         server_name localhost;               --
+--                                              --
+--         set $root /usr/share/nginx/html;     --
+--         root $root;                          --
+--                                              --
+--         location /index.html {               --
+--             # lua_code_cache off;            --
+--             set $default_lang "cz";          --
+--             set $ngx_html_path $root;        --
+--             rewrite_by_lua_file lang.lua;    --
+--         }                                    --
+--     }                                        --
+--                                              --
+--------------------------------------------------
+
+function scandir(directory)
+    local t = {}
+    for filename in io.popen('find "'..directory..'" -type d -mindepth 1 -maxdepth 1 -printf "%f\n"'):lines() do
+        table.insert(t, filename)
+    end
+    return t
+end
+
+function inTable(tbl, item)
+    for key, value in pairs(tbl) do
+        if value == item then return key end
+    end
+    return false
+end
+
+local default_lang = ngx.var.default_lang or "en"
+local folders = scandir(ngx.var.ngx_html_path)
+
+local lang_header = ngx.var.http_accept_language
+if ( lang_header == nil ) then
```

Comment:
this seems a little redundant since there is a default catch-all at the bottom. Maybe it could set `lang_header` to an empty string instead of nil?